src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -87,39 +114,128 @@ public JwtVendor(Settings settings, final LongSupplier timeProvider) {
     }
 
     /*
-    * The default configuration of this web key should be:
-    *   KeyType: OCTET
-    *   PublicKeyUse: SIGN
-    *   Encryption Algorithm: HS512
-    * */
-    static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
-        String signingKey = settings.get("signing_key");
+     * The default configuration of this web key should be:
+     *   KeyType: OCTET
+     *   PublicKeyUse: SIGN
+     *   Encryption Algorithm: HS512
+     * */
+    static JsonWebKey createJwkFromSettings(Settings settings) {
+        String algorithm = settings.get("algorithm");
+        Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
+
+        if (algorithm != null) {
+            return getSigningJwk(algorithm.toUpperCase(), settings);
+        } else if(!jwkSettings.isEmpty()) {
+            JsonWebKey jwk = new JsonWebKey();
+
+            for (String key : jwkSettings.keySet()) {
+                jwk.setProperty(key, jwkSettings.get(key));
+            }
 
-        if (!Strings.isNullOrEmpty(signingKey)) {
+            return jwk;
+        } else {
+            //try default
+            return getSigningJwk("HS512", settings);
+        }
+    }
 
-            JsonWebKey jwk = new JsonWebKey();
+    private static JsonWebKey getSigningJwk(String algorithm, Settings settings) throws IllegalArgumentException {
+        JsonWebKey jwk;
 
-            jwk.setKeyType(KeyType.OCTET);
-            jwk.setAlgorithm("HS512");
+        if (AlgorithmUtils.isOctet(algorithm)) {
+            jwk = new JsonWebKey();
             jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+            jwk.setKeyType(KeyType.OCTET);
+            String signingKey = Optional.ofNullable(settings.get("signing_key")).orElseThrow(() -> new IllegalArgumentException("Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."));
```

Comment:
We will want to document this; we can add a table or something with the supported keys and explain secret creation.